### PR TITLE
Update CocoaPods minimum version to 1.9

### PR DESCRIPTION
**Pull request by jmagman**
## Description

[CocoaPods 1.9](https://blog.cocoapods.org/CocoaPods-1.9.0-beta/) introduced support for xcframeworks.  It was released in [February 2020](https://github.com/CocoaPods/CocoaPods/releases/tag/1.9.0).  

- Update minimum to support Flutter being [distributed as an xcframework](https://github.com/flutter/flutter/issues/60109) to support [Apple Silicon simulators](https://github.com/flutter/flutter/issues/64502).
- Update recommended version to [1.10](https://blog.cocoapods.org/CocoaPods-1.10.0-beta/), which fixed many xcframework bugs and added Xcode 12 support.
- Drop `isCocoaPodsInitialized` which only mattered for versions < 1.8.

## Related Issues

https://github.com/flutter/flutter/pull/69809 is in 1.24.0-10.1.pre (currently dev and beta).  This change will be in `flutter-1.25-candidate.*` so the recommended minimum nudge should make it to stable before this harsher minimum requirement.

### CODE REVIEW

**Comment by jmagman** [comment]
> #69809 is in 1.24.0-10.1.pre (currently dev and beta). This change will be in `flutter-1.25-candidate.*` so the recommended minimum nudge should make it a stable before this harsher minimum requirement.

@christopherfujino this is true, right?

**Comment by jmagman** [comment]
Blocked by https://github.com/flutter/flutter/issues/69829

**Comment by jmagman** [comment]
I think I upgraded all the agents.  Let's merge, I'll revert if something breaks post-submit.

File: packages/flutter_tools/lib/src/macos/cocoapods_validator.dart [hunk 1]
```diff
@@ -30,12 +30,7 @@ class CocoaPodsValidator extends DoctorValidator {
 
     ValidationType status = ValidationType.installed;
     if (cocoaPodsStatus == CocoaPodsStatus.recommended) {
-      if (await _cocoaPods.isCocoaPodsInitialized) {
-        messages.add(ValidationMessage(_userMessages.cocoaPodsVersion(await _cocoaPods.cocoaPodsVersionText)));
-      } else {
-        status = ValidationType.partial;
-        messages.add(ValidationMessage.error(_userMessages.cocoaPodsUninitialized(noCocoaPodsConsequence)));
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/macos/cocoapods_validator.dart:37]: can we also delete this user message?

**Review by christopherfujino — APPROVED** [review]
nit about removing the user message, otherwise this is great! LGTM

File: packages/flutter_tools/lib/src/macos/cocoapods_validator.dart (hunk 1 above)
Comment (jmagman) [reply packages/flutter_tools/lib/src/macos/cocoapods_validator.dart:37]: I missed that, good eye.

**Comment by blasten** [comment]
This change appears to be causing the issues seen in https://github.com/flutter/plugins/pull/3342/checks?check_run_id=1573661704. Is this considered a breaking change?

**Comment by jmagman** [comment]
@blasten it's not a breaking change, I just had to update plugin's CI to an updated version of CocoaPods.  https://github.com/flutter/plugins/pull/3347

**Pull request merged**